### PR TITLE
docs: add better documentation on dependencies

### PR DESCRIPTION
**Pull request by ahal**
After needing to explain how to write a dependency, I realized the docs on this topic were sorely lacking.

### CODE REVIEW

**Comment by ahal** [comment]
Rendered version can be found here:
https://taskcluster-taskgraph--145.org.readthedocs.build/en/145/concepts/task-graphs.html#dependencies

File: docs/concepts/task-graphs.rst [hunk 1]
```diff
@@ -118,33 +118,111 @@ Dependencies
 ------------
 
 Dependencies between tasks are represented as labeled edges in the task graph.
-For example, a test task must depend on the build task creating the artifact it
-tests, and this dependency edge is named 'build'. The task graph generation
-process later resolves these dependencies to specific task ids.
+They are specified via the ``dependencies`` key which is an object of the form
+``{ "<edge>": "<label>"}``. The ``edge`` is an arbitrary name which can be used
+to refer to the dependency later on. The ``label`` is the task label of the
+dependency.
```
Comment (escapewindow) [inline docs/concepts/task-graphs.rst:124]: I may be misremembering, but the value can also be a Task object, no?

**Review by escapewindow — APPROVED** [review]
Thanks! This definitely needed clarification.

File: docs/concepts/task-graphs.rst (hunk 1 above)
Comment (ahal) [reply docs/concepts/task-graphs.rst:124]: I'm not sure.. The `task_description_schema` allows it, but I'm failing to find where we translate from `Task` object -> label (as the latter is definitely what's expected). Maybe it used to be supported but got removed at some point?

**Pull request merged**